server/datastore/mysql/software_installers_test.go

```diff
@@ -1776,3 +1777,233 @@ func testBatchSetSoftwareInstallersScopedViaLabels(t *testing.T, ds *Datastore)
+	require.NotNil(t, noTeamPolicies[0].TeamID)
+	require.Equal(t, fleet.PolicyNoTeamID, *noTeamPolicies[0].TeamID)
+
+	// Test deb.
+	installerID3, _, err := ds.MatchOrCreateSoftwareInstaller(ctx, &fleet.UploadSoftwareInstallerPayload{
+		InstallerFile:    tfr1,
+		Extension:        "deb",
+		StorageID:        "storage3",
+		Filename:         "barfoo1",
+		Title:            "Barfoo",
+		Version:          "1.0",
+		Source:           "deb_packages",
+		UserID:           user1.ID,
+		TeamID:           &team2.ID,
+		AutomaticInstall: true,
+		PackageIDs:       []string{"id1"},
+		ValidatedLabels:  &fleet.LabelIdentsWithScope{},
+	})
+	require.NoError(t, err)
+
+	team2Policies, _, err := ds.ListTeamPolicies(ctx, team2.ID, fleet.ListOptions{}, fleet.ListOptions{})
+	require.NoError(t, err)
+	require.Len(t, team2Policies, 1)
+	require.Equal(t, "[Install software] Barfoo (deb)", team2Policies[0].Name)
+	require.Equal(t, `SELECT 1 WHERE EXISTS (
+	SELECT 1 FROM os_version WHERE platform = 'rhel'
+) OR EXISTS (
+	SELECT 1 FROM deb_packages WHERE name = 'Barfoo'
+);`, team2Policies[0].Query)
+	require.Equal(t, `Policy triggers automatic install of Barfoo on each host that's missing this software.
+Software won't be installed on Linux hosts with Debian-based distributions because this policy's query is written to always pass on these hosts.`, team2Policies[0].Description)
+	require.Equal(t, "linux", team2Policies[0].Platform)
+	require.NotNil(t, team2Policies[0].SoftwareInstallerID)
+	require.Equal(t, installerID3, *team2Policies[0].SoftwareInstallerID)
+	require.NotNil(t, team2Policies[0].TeamID)
+	require.Equal(t, team2.ID, *team2Policies[0].TeamID)
+
+	// Test rpm.
+	installerID4, _, err := ds.MatchOrCreateSoftwareInstaller(ctx, &fleet.UploadSoftwareInstallerPayload{
+		InstallerFile:    tfr1,
+		Extension:        "rpm",
+		StorageID:        "storage4",
+		Filename:         "barzoo1",
+		Title:            "Barzoo",
+		Version:          "1.0",
+		Source:           "rpm_packages",
+		UserID:           user1.ID,
+		TeamID:           &team2.ID,
+		AutomaticInstall: true,
+		PackageIDs:       []string{"id1"},
+		ValidatedLabels:  &fleet.LabelIdentsWithScope{},
+	})
+	require.NoError(t, err)
+
+	team2Policies, _, err = ds.ListTeamPolicies(ctx, team2.ID, fleet.ListOptions{}, fleet.ListOptions{})
+	require.NoError(t, err)
+	require.Len(t, team2Policies, 2)
+	require.Equal(t, "[Install software] Barzoo (rpm)", team2Policies[1].Name)
+	require.Equal(t, `SELECT 1 WHERE EXISTS (
+	SELECT 1 FROM os_version WHERE platform != 'rhel'
+) OR EXISTS (
+	SELECT 1 FROM rpm_packages WHERE name = 'Barzoo'
+);`, team2Policies[1].Query)
+	require.Equal(t, `Policy triggers automatic install of Barzoo on each host that's missing this software.
+Software won't be installed on Linux hosts with RPM-based distributions because this policy's query is written to always pass on these hosts.`, team2Policies[1].Description)
```

Comment:
See above

Reply:
Thanks for catching this double error :)